### DISCUSSION TITLE
local development environment with docker

### DISCUSSION THREAD

**Discussion by ffjlabo** [pipe-cd/pipecd#4694] [Ideas]
Setting up an environment so that development can be done entirely in a local environment might make it easier for contributors to participate. WDYT?

If the following can work on docker-compose, users only prepare the environment where piped works.
- git server
- control plane

(It might be good to make piped work on docker-compose )

**Comment by khanhtc1202** [comment]
Do you mean this? 🤔 
https://github.com/pipe-cd/pipecd/pull/4139

**Comment by khanhtc1202** [comment]
Also, we have how to run pipecd control plane locally docs already 😄 
https://github.com/pipe-cd/pipecd/tree/master/cmd/pipecd#how-to-run-control-plane-locally

**Comment by t-kikuc** [comment]
git:

`files:///Users/PATH/TO/LOCAL/REPO` will work locally, although the behavior might be completely correct.
